recipes/imgui-bundle/singel_patch.patch

```diff
@@ -0,0 +1,854 @@
+ 
+-target_link_libraries(${target_name} PUBLIC
+-    imgui
+-    immvision_gl_loader
+-    opencv_core
+-    opencv_imgproc
+-    opencv_imgcodecs
+-    )
+-
++target_link_libraries(${target_name} PUBLIC imgui immvision_gl_loader)
++target_link_libraries(${target_name} PRIVATE opencv_core)
+diff --git a/external/immvision/immvision/src/immvision_gl_loader/CMakeLists.txt b/external/immvision/immvision/src/immvision_gl_loader/CMakeLists.txt
+index 14df249..59d9540 100644
+--- a/external/immvision/immvision/src/immvision_gl_loader/CMakeLists.txt
++++ b/external/immvision/immvision/src/immvision_gl_loader/CMakeLists.txt
+@@ -28,11 +28,12 @@ target_include_directories(immvision_gl_loader PUBLIC $<BUILD_INTERFACE:${CMAKE_
+ 
+ if (NOT EMSCRIPTEN)
+     find_package(OpenGL REQUIRED)
+-    target_link_libraries(immvision_gl_loader PUBLIC OpenGL::GL)
++    target_link_libraries(immvision_gl_loader PRIVATE OpenGL::GL)
+ endif()
+ 
+ if (IMMVISION_USE_GLAD)
+-    target_link_libraries(immvision_gl_loader PUBLIC glad_imm)
++    target_link_libraries(immvision_gl_loader PRIVATE glad_imm)
++    target_include_directories(immvision_gl_loader PUBLIC $<BUILD_INTERFACE:${GLAD_DIR}/include>)
+     target_compile_definitions(immvision_gl_loader PUBLIC IMMVISION_USE_GLAD)
+ elseif(IMMVISION_USE_GLES3)
+     target_compile_definitions(immvision_gl_loader PUBLIC IMMVISION_USE_GLES3)
+diff --git a/external/immvision/immvision/src_all_in_one/immvision/CMakeLists.txt b/external/immvision/immvision/src_all_in_one/immvision/CMakeLists.txt
+index 06a6bd4..3b91d88 100644
+--- a/external/immvision/immvision/src_all_in_one/immvision/CMakeLists.txt
++++ b/external/immvision/immvision/src_all_in_one/immvision/CMakeLists.txt
+@@ -6,5 +6,5 @@ target_include_directories(${target_name} PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/..)
+ target_link_libraries(${target_name} PUBLIC
+     imgui
+     immvision_gl_loader
+-    opencv_core opencv_imgproc opencv_highgui opencv_imgcodecs
+-    )
++)
++target_link_libraries(${target_name} PRIVATE opencv_core opencv_imgproc opencv_highgui opencv_imgcodecs)
+diff --git a/imgui_bundle_cmake/imgui_bundle_build_lib.cmake b/imgui_bundle_cmake/imgui_bundle_build_lib.cmake
+index c7fa439b..50d12f2e 100644
+--- a/imgui_bundle_cmake/imgui_bundle_build_lib.cmake
++++ b/imgui_bundle_cmake/imgui_bundle_build_lib.cmake
+@@ -34,17 +34,6 @@ function(ibd_force_freetype_static_for_python)
+     endif()
+ endfunction()
+ 
+-# See https://github.com/pthom/imgui_bundle/issues/261 and https://chatgpt.com/share/66ffb718-6408-8004-be5b-9e74064a8709
+-function(ibd_disable_system_libraries_for_python)
+-    if(IMGUI_BUNDLE_BUILD_PYTHON AND
+-       (NOT "$ENV{CMAKE_TOOLCHAIN_FILE}" MATCHES "vcpkg" AND NOT DEFINED CONAN_BUILD)  # Not using vcpkg or conan
+-       )
+-        message(STATUS "ibd_disable_system_libraries_for_python: Disabling system libraries for python bindings")
+-        set(CMAKE_FIND_FRAMEWORK NEVER CACHE STRING "" FORCE)
```

Comment:
ps @pthom  I think things like "FORCE" are really hurtful for system integrators. I don't really think any of this is necessary and is typically handled by cibuildwheel (or at least I thought it was)

Reply:
Sorry for that. I’ll study if it can be removed.
I remember adding it because skbuild + pip was still catching some local libraries when building wheels that are to be distributed.

Reply:
yeah i've struggled with the flags before. so long as they can be overwritten it is typically ok.